crates/services/p2p/src/codecs.rs

```diff
@@ -1,19 +1,24 @@
+pub mod bounded;
 pub mod postcard;
+pub mod unbounded;
 
-use crate::{
-    gossipsub::messages::{
-        GossipTopicTag,
-        GossipsubBroadcastRequest,
-        GossipsubMessage,
-    },
-    request_response::messages::{
-        RequestMessage,
-        V2ResponseMessage,
-    },
-};
+use crate::gossipsub::messages::GossipTopicTag;
 use libp2p::request_response;
+use serde::{
+    Deserialize,
+    Serialize,
+};
 use std::io;
 
+trait DataFormatCodec {
+    type Error;
+    fn deserialize<'a, R: Deserialize<'a>>(
+        encoded_data: &'a [u8],
```

Comment:
This could be generalized to a `impl std::io::Read`.

Reply:
I'm not sure I follow completely here, do you mean changing the type of `econded_data`? If we do so, we need to allocate a second buffer where to read the value into. Or do you have something else in mind?

Reply:
> I'm not sure I follow completely here, do you mean changing the type of econded_data? 
Yes, the suggestion taking `encoded_data: impl std::io::Read + 'a`, instead of explicitly taking a slice here since it allows users to pass in any reader, which is more flexible. However, I see that we use `AsyncRead` and not `Read` for the network communication which would be a bit more messy so let's not do this.
> If we do so, we need to allocate a second buffer where to read the value into. Or do you have something else in mind?
That shouldn't be necessary since `&[u8]` also implements `Read`, so all current calls should still work. Or perhaps I'm misunderstanding you here.